src/LineBot/Actions/DateTimePickerAction.cs

```diff
@@ -0,0 +1,195 @@
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the string returned via webhook in the postback.data property of the <see cref="IPostback"/> event.
+        /// <para>Max: 300 characters.</para>
+        /// </summary>
+        [JsonProperty("data")]
+        public string Data
+        {
+            get
+            {
+                return _data;
+            }
+
+            set
+            {
+                if (string.IsNullOrWhiteSpace(value))
+                    throw new InvalidOperationException("The data cannot be null or whitespace.");
+
+                if (value.Length > 300)
+                    throw new InvalidOperationException("The data cannot be longer than 300 characters.");
+
+                _data = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the action mode for the datetimepicker.
+        /// </summary>
+        [JsonProperty("mode")]
+        [JsonConverter(typeof(EnumConverter<DateTimePickerMode>))]
+        public DateTimePickerMode Mode
+        {
+            get
+            {
+                return _mode;
+            }
+
+            set
+            {
+                _mode = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the initial datetime for the datetimepicker.
+        /// </summary>
+        [JsonProperty("initial")]
+        public DateTime? Initial
+        {
+            get
+            {
+                return _initial;
+            }
+
+            set
+            {
+                _initial = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the min datetime for the datetimepicker.
+        /// </summary>
+        [JsonProperty("min")]
+        public DateTime? Min
+        {
+            get
+            {
+                return _min;
+            }
+
+            set
+            {
+                if (Max.HasValue && value > Max.Value)
+                    throw new InvalidOperationException("The min must be less than the max.");
+
+                _min = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the max datetime for the datetimepicker.
+        /// </summary>
+        [JsonProperty("max")]
+        public DateTime? Max
+        {
+            get
+            {
+                return _max;
+            }
+
+            set
+            {
+                if (Min.HasValue && value < Min.Value)
+                    throw new InvalidOperationException("The max must be greater than the min.");
+
+                _max = value;
+            }
+        }
+
+        void IAction.Validate()
+        {
+            if (_label == null)
+                throw new InvalidOperationException("The label cannot be null.");
+
+            if (_data == null)
+                throw new InvalidOperationException("The data cannot be null.");
+
+            if (Min.HasValue && Max.HasValue && Min.Value > Max.Value)
```

Comment:
I don't think you need to check the date here again? This is already validated in the properties.